spark/src/main/java/org/apache/zeppelin/spark/SparkSqlInterpreter.java

```diff
@@ -114,7 +170,11 @@ public InterpreterResult interpret(String st, InterpreterContext context) {
       // to    def sql(sqlText: String): DataFrame (1.3 and later).
       // Therefore need to use reflection to keep binary compatibility for all spark versions.
       Method sqlMethod = sqlc.getClass().getMethod("sql", String.class);
-      rdd = sqlMethod.invoke(sqlc, st);
+      String interpolateResult = interpolateVariable(st);
+      if (interpolateResult.startsWith("ERROR:"))
```

Comment:
using a String as status return is a bit odd, how about returning a bool and the result separately?